src/identity_tree.rs

```diff
@@ -263,6 +263,9 @@ impl BasicTreeOps for TreeVersionData<lazy_merkle_tree::Derived> {
     fn update(&mut self, leaf_index: usize, element: Hash) {
         let updated_tree = self.tree.update(leaf_index, &element);
 
+        if self.tree.root() == updated_tree.root() {
```

Comment:
I added it here because of recreating tree on startup. It is hard to easily select from database proper groups of commitments. There are three options:
1. As done change the code to not apply updates that don't change the tree at all. Why? Because then we are peeking changes to create a batch that are not changing anything and we can't put such a batch in database (unique/fk constraints).
2. Instead of doing it here we can peek the "no-changing" changes and just not save them in database.
3. Add additional internal status just for this "batched but not processed yet" situation.
4. Do more queries on startup but may use much more memory as we are loading almost all commitments. :/